dataRetriever/txpool/shardedTxPool.go

```diff
@@ -270,6 +285,13 @@ func (txPool *shardedTxPool) removeTxFromAllShards(txHash []byte) {
 	}
 }
 
+// ForgetAllAccountNoncesInMempool forgets all account nonces in the mempool
+func (txPool *shardedTxPool) ForgetAllAccountNoncesInMempool() {
+	selfShardID := strconv.Itoa(int(txPool.selfShardID))
```

Comment:
Question: Is this robust enough?

Reply:
can't you do at txPool constructor level ?

Reply:
Good point, will do.